packages/core/src/components/ColorPalette/EditVariantModal.tsx

```diff
@@ -74,7 +74,15 @@ export function EditVariantModal({
     }
     // Remove error so it doesn't persist...
     setError(null)
-    onUpdateVariant(variant)
+    // catch any thrown errors on save
+    try {
+      onUpdateVariant(variant)
+    } catch (err) {
+      if (err instanceof Error) {
+        setError(err.message)
+        return
+      }
```

Comment:
instead of this and throwing an sett, can you just do 

```javascript
 if (newVariant.name in colorData) {
         setError('This name already exists, please choose a different name.')
 }
 ```

Reply:
i don't think you have access to `colorData` in `EditVariantModal.tsx` so you cannot check before submitting 

also in `ColorDisplay.tsx`  don't have access to the error state because its initialization is in `EditVariantModal.tsx`  so you cannot change it in the parent

Reply:
but i can see why you don't want to throw an error, I guess I can move the `error` state from the child to the parent component and pass it as prop to the child
- move `error` state to `ColorDisplay` to be controlled by the parent
- `EditVariantModal` -the child here- would take another prop which is the error state

Reply:
Ok, I guess it's fine as is.

Reply:
Thanks for your help @nagy-nabil! 🎊